src/pages/actions.tsx

```diff
@@ -0,0 +1,476 @@
+                  !activeUser.notifyElectionReminders &&
+                  !activeUser.startedChallenge) ||
+                  (!activeUser.registeredVoter &&
+                    activeUser.notifyElectionReminders &&
+                    !activeUser.startedChallenge)) && (
+                  <button
+                    type="button"
+                    className={styles.button_link}
+                  >
+                    Share about your action
+                  </button>
+                )}
+
+                {(((activeUser.registeredVoter || activeUser.notifyElectionReminders) &&
+                  activeUser.startedChallenge) || (activeUser.registeredVoter && 
+                  activeUser.notifyElectionReminders && !activeUser.startedChallenge)) && (
+                    <button
+                      type="button"
+                      className={styles.button_link}
+                    >
+                      Share about your actions
+                    </button>
+                  )}
+
+                {activeUser.startedChallenge && (
+                  <Link href="/signin" className={styles.links}>
+                    See your challenge
+                  </Link>
+                )}
+              </div>
+            </div>
+            
+          </div>
+          :
+          <div>
+            {inviterInfo.finishedChallenge && !choosenReward.logo && rewardsAvailable && (
+              <button
+                type="button"
+                className={styles.first_button}
+                onClick={() => router.push(`/choosereward`)}
+              >
+                <span>Choose a Reward</span>
+              </button>
+            )}
+            
+            <button
+              type="button"
+              className={
+                inviterInfo.finishedChallenge && rewardsAvailable 
+                  ? styles.second_button
+                  : styles.first_button
+              }
+              onClick={() => router.push(`/signin`)}
+            >
+              <span>See Your Challenge</span>
+            </button>
+
+            <button
+              type="button"
+              className={styles.second_button}
+            >
+              <span>Share About Your Actions</span>
+            </button>
+
+            {playerStateInfo.onlinereg && (
+              <button
+                type="button"
+                onClick={() => window.open(playerStateInfo.voteregsite, "_blank")}
+                className={styles.second_button}
+              >
+                <span>Go to state website</span>
+              </button>
+            )}
+
+            <button
+              type="button"
+              className={styles.second_button}
+              onClick={() => {setShowVoterFormModal(true)}}
+            >
+              <span>Get your registration form again</span>
+            </button>
+          </div>
+          }
+        </section>
+      </div>
+      
+      
+      {showVoterFormModal && (
```

Comment:
I'm having trouble testing this component. The two tests I wrote for this specific part error because in the tests I try to open the modal and its calling the PopModal component.

Reply:
Hello Naim,

Thank you for bringing this up. The issue is that, although the <dialog /> element is supported in modern browsers, and it has been used here for accessibility reasons, jsdom does not currently support it. Please see this thread for more information: https://github.com/jsdom/jsdom/issues/3294 

A workaround is to add the showModal property to HTMLDialogElement.prototype prior to executing the tests. 

A simple example of this can be found in the snapshot file for the error-modal, which can be found in src/__tests__/components/error-modal.

I also like this implementation: https://github.com/jsdom/jsdom/issues/3294#issuecomment-1268330372 

Please let me know if there are any other difficulties I can help address. Thank you!

Reply:
Thanks this worked.